src/opencl/utils.rs

```diff
@@ -107,17 +107,16 @@ fn build_device_list() -> HashMap<Brand, Vec<Device>> {
                         devices
                             .into_iter()
                             .filter(|d| {
-                                // Only return available devices.
-                                if !d.is_available().unwrap_or(false) {
-                                    return false;
+                                if let Ok(vendor) = d.vendor() {
+                                    match vendor.as_str() {
+                                        // Only use devices from the accepted brands ...
+                                        "AMD" | "NVIDIA" => {
+                                            // ... which are available.
```

Comment:
are you certain they will be always uppercase?

Reply:
They should always be the same, but the NVIDIA one was wrong elsewhere. I made them constants so we can get them right once and use as needed.